src/main/java/com/kodilla/ecommercee/controller/CartController.java

```diff
@@ -2,52 +2,66 @@
-        //post to repository//
-        return ResponseEntity.ok().build();
+    @PostMapping(consumes = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<String> createCart(@RequestBody CartDto cartDto) {
+        return new ResponseEntity<>("New cart was created for userid: " + cartDto.getUserId(), HttpStatus.OK);
     }
     
-    @GetMapping(value = "/{userId}/{cartId}")
-    public ResponseEntity<CartDto> getCart(@PathVariable Long userId, @PathVariable Long cartId) {
-        //get from repository//
-        //map to Dto//
-        CartDto currentCard = new CartDto(userId, cartId, new ArrayList<>());
-        return ResponseEntity.ok(currentCard);
+    @GetMapping("/products")
+    public ResponseEntity<List<CartItemDto>> getCart(@RequestParam long cartId) {
+        List<CartItemDto> cartItemDtos = new ArrayList<>();
+        cartItemDtos.add(new CartItemDto(
+                1L,"product1","description1", new BigDecimal(100),4L));
+        cartItemDtos.add(new CartItemDto(
+                2L,"product2","description2", new BigDecimal(200),3L));
+        return ResponseEntity.ok(cartItemDtos);
     }
 
-    @PostMapping(value = "{cartId}")
-    public ResponseEntity<Void> addCartItem(@PathVariable Long cartId, @RequestBody CartItemDto cartItemDto) {
-        //post item into Cart//
-        return ResponseEntity.ok().build();
+    @PutMapping("/add/{cartId}")
+    public ResponseEntity<String> addProductToCart(@PathVariable long cartId, @RequestParam long productId) {
+        // below should be executed in DbService
+        // Cart cart = findById(cartId)
+        // cart.getProductList.add(findProductById(productId);
+        // above is just rought representation of execution.
+        return new ResponseEntity<>("Product: " + productId + " was added to cart: " + cartId, HttpStatus.OK);
     }
 
-    @DeleteMapping(value = "/deleteItem/{cartItemId}")
-    public ResponseEntity<Void> removeCartItem(@PathVariable Long cartItemId,@RequestParam Long cartId) {
-        return ResponseEntity.ok().build();
+    @PutMapping("/remove{cartId}")
+    public ResponseEntity<String> removeProductToCart(@PathVariable long cartId, @RequestParam long productId) {
+        // below should be executed in DbService
+        // Cart cart = findById(cartId)
+        // cart.getProductList.remove(findProductById(productId);
+        // above is just rought representation of execution.
+        return new ResponseEntity<>("Product: " + productId + " was removed from cart: " + cartId, HttpStatus.OK);
     }
 
-    @PostMapping(value = "/{cartId}/order")
-    public ResponseEntity<CartItemDto> createOrder(@PathVariable Long cartId) {
-        //Method will return OrderDto when it is ready//
-        //Get current cart from repository//
-        //Map current cart to OrderDto//
-        return ResponseEntity.ok().build();
+    @PostMapping("/createOrder")
+    public ResponseEntity<OrderDto> createOrder(@RequestParam long cartId) {
```

Comment:
I would try:
    @PostMapping("/createOrder/{cartId}")
    public ResponseEntity<OrderDto> createOrder(@PathVariable Long cartId) {

Reply:
With @RequestParam you can have it like this: http://localhost:8080/carts/createOrder?cartId=5. You put your parameter cartId after ?cartID=. I don't know what is better @PathVariable or @RequestParam.